src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -133,15 +132,15 @@ private void CreateStreamAndStartWatcher()
             if (String.IsNullOrEmpty(_directory) == false)
             {
                 // Get the path to watch and verify we created the CFStringRef
-                CFStringRef path = Interop.CoreFoundation.CFStringCreateWithCString(_directory);
-                if (path == IntPtr.Zero)
+                SafeCreateHandle path = Interop.CoreFoundation.CFStringCreateWithCString(_directory);
+                if (path.IsInvalid)
                 {
                     throw Interop.GetExceptionForIoErrno(Marshal.GetLastWin32Error(), _directory, true);
                 }
 
                 // Take the CFStringRef and put it into an array to pass to the EventStream
-                CFArrayRef arrPaths = Interop.CoreFoundation.CFArrayCreate(new CFStringRef[1] { path }, 1);
-                if (arrPaths == IntPtr.Zero)
+                SafeCreateHandle arrPaths = Interop.CoreFoundation.CFArrayCreate(new CFStringRef[1] { path.DangerousGetHandle() }, 1);
+                if (arrPaths.IsInvalid)
```

Comment:
Won't dispose be called when the exception is thrown? The path and arrPaths variables should go out of scope, have no references, and be GC'd then (whenever the GC runs next). Is that not correct?